File: Facebook-Backend/src/main/java/com/project/facebook/responses/profile/ProfileHeaderResponse.java
package com.project.facebook.responses.profile;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Builder
@Getter
@Setter
public class ProfileHeaderResponse {
    @JsonProperty("avatar")
    private String avatar;

    @JsonProperty("cover_photo")
    private String coverPhoto;


    @JsonProperty("fullname")
    private String fullname;

    @JsonProperty("total_friends")
    private int totalFriends;

    @JsonProperty("mutual_friends")
    private int mutualFriends;

    @JsonProperty("is_friends")
    private boolean isFriends;

    @JsonProperty("avatar_friends")
    private List<ProfileAvatarFriendsResponse> avatarFriends;
}

File: Facebook-Frontend/src/app/shared/responses/profile/profile-header.response.ts
import { ImageResponse } from '../common/image.response';

export class ProfileHeaderResponse {
  profile_id: number = 0;
  avatar: string = '';
  cover_photo: string = '';
  fullname: string = '';
  total_friends: number = 0;
  mutual_friends: number = 0;
  is_friends: boolean = false;
  is_online: boolean = false;
  avatar_friends: ImageResponse[] = [];
}
